src/Reflection/ReflectionFunctionAbstract.php

```diff
@@ -152,4 +210,143 @@ public function getFileName()
     {
         return $this->filename;
     }
+
+    /**
+     * Is this function a closure?
+     *
+     * Note - we cannot reflect on closures at the moment (as there is no PHP
+     * source code we can access).
+     *
+     * @see https://github.com/Roave/BetterReflection/issues/37
+     * @return bool
+     */
+    public function isClosure()
+    {
+        return false;
+    }
+
+    /**
+     * Is this function deprecated?
+     *
+     * Note - we cannot reflect on internal functions (as there is no PHP source
+     * code we can access. This means, at present, we can only EVER return false
+     * from this function.
+     *
+     * @see https://github.com/Roave/BetterReflection/issues/38
+     * @return bool
+     */
+    public function isDeprecated()
+    {
+        return false;
+    }
+
+    /**
+     * Is this an internal function?
+     *
+     * Note - we cannot reflect on internal functions (as there is no PHP source
+     * code we can access. This means, at present, we can only EVER return false
+     * from this function.
+     *
+     * @see https://github.com/Roave/BetterReflection/issues/38
+     * @return bool
+     */
+    public function isInternal()
+    {
+        return false;
+    }
+
+    /**
+     * Is this a user-defined function (will always return the opposite of
+     * whatever isInternal returns).
+     *
+     * @return bool
+     */
+    public function isUserDefined()
+    {
+        return !$this->isInternal();
+    }
+
+    /**
+     * Check if the function has a variadic parameter
+     *
+     * @return bool
+     */
+    public function isVariadic()
+    {
+        $parameters = $this->getParameters();
+
+        foreach ($parameters as $parameter) {
+            if ($parameter->isVariadic()) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Recursively search an array of statements (PhpParser nodes) to find if a
+     * yield expression exists anywhere.
+     *
+     * @param \PhpParser\Node[] $statements
+     * @return bool
+     */
+    private function checkStatementsForYield(array $statements)
+    {
+        foreach ($statements as $stmt) {
+            if ($stmt instanceof YieldNode) {
+                return true;
+            }
+            if (isset($stmt->stmts) && is_array($stmt->stmts) && count($stmt->stmts)) {
+                if ($this->checkStatementsForYield($stmt->stmts)) {
+                    return true;
+                }
```

Comment:
Thanks again @nikic - I hope you don't mind but I borrowed your test cases to check we've covered all the various yields (in PHP5 at least), I've credited you [here](https://github.com/Roave/BetterReflection/pull/40/files#diff-89b993f0e9c9d3c15b60bd5f3700c999R65), let me know if you object or anything!